byron/semantics/executable-spec/src/Control/State/Transition/Generator.hs

```diff
@@ -257,17 +255,17 @@ genTraceOfLength
   -- generate a signal.
   -> Gen (Trace s)
 genTraceOfLength aTraceLength profile env st0 aSigGen =
-  mapGenT (TreeT . interleaveSigs . runTreeT) $ loop aTraceLength st0 []
+  Manual.fromManual $ fmap interleaveSigs $ loop aTraceLength st0 []
   where
     loop
       :: Word64
       -> State s
       -> [(State s, TreeT (MaybeT Identity) (Signal s))]
-      -> Gen [(State s, TreeT (MaybeT Identity) (Signal s))]
+      -> Manual [(State s, TreeT (MaybeT Identity) (Signal s))]
     loop 0 _ acc = pure acc
-    loop d sti acc = do
+    loop !d sti acc = do
```

Comment:
This seems odd to me! In general we want to be strict in the accumulator, to avoid concatenation thunks. And it would seem that should be the case here. Whereas `d` surely gets evaluated on each call to loop in order to check whether it's `0`, so there's no major advantage in making it strict.

Perhaps I'm missing something, in which case I'd welcome a comment on this function explaining the strictness precisely, since it doesn't seem obvious :-)

Reply:
"There is no major advantage in making it strict" -- this isn't quite accurate, is it? `d` _is_ strict, due to the pattern match the line before. The question is, is there an advantage on sticking in the bang? I would say, it's useful to add a bang, even if technically superfluous, if indeed it is _important_ that this variable is strict: it states intent.

However, making the _accumulator_ strict here I think is actively misleading: we are accumulating a _list_. Adding a bang here would achieve precisely nothing, because the list will already _be_ in WHNF; adding a bang would suggest that something is being forced, but it really isn't.